Wazuh Cloud: Documentation for the new pricing model (#6258)
wazuh/wazuh-documentation: source/cloud-service/your-environment/settings.rst

.. Copyright (C) 2015, Wazuh, Inc.

.. meta::
  :description: Learn about Wazuh Cloud settings. 

.. _cloud_your_environment_settings:

Settings
========

Every cloud environment is configured based on specific settings that define its limitations, capabilities, and pricing. We offer six available settings, including four basic settings and two advanced settings. The advanced settings are automatically calculated based on the basic settings but can be modified if desired.

To monitor the behavior of your environment and check if the configured values of the settings are being reached, see the :doc:`Monitor usage section </cloud-service/your-environment/monitor-environment-usage>`. section.

Understanding environment settings
----------------------------------

Below, we provide a detailed explanation of each setting.

.. _cloud_settings_active_agents:

Active agents
^^^^^^^^^^^^^

> This basic setting defines the maximum number of active Wazuh agents that the environment can support. Please note that while it is possible to register an unlimited number of Wazuh agents, the number of active agents is limited by this setting.

Reviewer: ```suggestion
This basic setting sets the maximum count of active Wazuh agents that the environment can support. Please note that while registering an unlimited number of Wazuh agents is possible, the active agent count is limited by this setting.
```

> If the maximum number of active agents is reached, the environment may start to malfunction, causing instability with agent connections. While the system can tolerate temporarily exceeding the limit of active agents, appropriate measures will be taken if the situation persists.

Reviewer: ```suggestion
If the maximum number of active agents is reached, the environment might start to malfunction, causing instability with agent connections. Although the system can temporarily handle exceeding the active agent limit, appropriate measures will be taken if the situation persists.
```

> 

Reviewer: ```suggestion
```

.. _cloud_settings_indexed_data:

Indexed data
^^^^^^^^^^^^

> The indexed data, previously known as hot storage, includes the data available on the Wazuh WUI, which corresponds to the information indexed by Wazuh. This information becomes searchable and analyzable as soon as Wazuh ingests and indexes the events sent by the agents.

Reviewer: ```suggestion
The indexed data was previously known as hot storage. It includes the data available on the Wazuh dashboard, which corresponds to the information indexed by Wazuh. This information becomes searchable and analyzable as soon as Wazuh ingests and indexes the events sent by the agents.
```

Two settings define the behavior of the indexed data:

> - Indexed data time: It determines the maximum duration for which data remains indexed. This is a basic setting.

Reviewer: ```suggestion
-  **Indexed data time**: This basic setting defines the maximum duration for which data remains indexed.
```

> - Indexed data size: It defines the maximum size, in bytes, of the indexed data. This setting is equivalent to what was previously called a "tier." This is an advanced setting, and the interface provides a suggestion when selecting the Indexed data time.

Reviewer: ```suggestion
-  **Indexed data size**: This advanced setting defines the maximum size, in bytes, of the indexed data. it was formerly referred to as a "tier". The interface offers a suggestion when selecting the Indexed data time.
```

> - Indexed data size: It defines the maximum size, in bytes, of the indexed data. This setting is equivalent to what was previously called a "tier." This is an advanced setting, and the interface provides a suggestion when selecting the Indexed data time.

Reviewer: It's not quite clear what the "... suggestion when selecting the Indexed data time." is all about.

> Data remains indexed until either the indexed data time or the indexed data size is reached. In other words, once either of the settings' values is reached, data rotation will occur (removing the oldest data) until the settings' conditions are met.

Reviewer: ```suggestion
Data remains indexed until either the indexed data time or the indexed data size is reached. Upon reaching either setting value, data rotation initiates, removing the oldest data, until the conditions of these settings are met.
```

.. _cloud_settings_archived_data:

Archived data
^^^^^^^^^^^^^

> This basic setting, previously known as cold storage, determines the duration for which the analyzed data generated by Wazuh is stored in an AWS S3 bucket for long-term storage purposes. Unlike the indexed data, this data is not searchable or analyzable. It simply consists of a collection of compressed files.

Reviewer: ```suggestion
This basic setting, previously known as cold storage, defines the duration for which the analyzed data generated by Wazuh is stored in an AWS S3 bucket for long-term storage purposes. Unlike the indexed data, this data isn't searchable or analyzable. It simply consists of a collection of compressed files.
```

> When the specified time is reached, data outside that time range will be deleted.

Reviewer: ```suggestion
When the specified time is reached, data beyond that time range will be deleted.
```

.. _cloud_support_type:

Support type
^^^^^^^^^^^^

This setting indicates whether the support level is premium or standard.

.. _cloud_settings_data_ingestion_rate:

Data ingestion rate
^^^^^^^^^^^^^^^^^^^

> It represents the maximum number of events per second (EPS) that the environment can analyze. This is an advanced setting, and the interface provides a suggestion when selecting the Active agents setting.

Reviewer: ```suggestion
It represents the maximum number of events per second (EPS) that the environment can analyze. This is an advanced setting, and the interface provides a suggestion when selecting the ``Active agents`` setting.
```

> If the ingestion rate is exceeded, events will start to queue. However, if the queue becomes full, the incoming events will be discarded, which may lead to potential event loss. The queuing mechanism is automatically managed by the cloud service, ensuring optimal resource utilization.

Reviewer: ```suggestion
If the ingestion rate is exceeded, events begin queuing. However, if the queue becomes full, wazuh discards the incoming events, which might lead to event loss. The cloud service automatically manages the queuing mechanism, ensuring optimal resource utilization.
```

> The environment will be configured with the `limits eps option <https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#limits>`_ using the following parameters:

Reviewer: ```suggestion
The environment is configured with the `limits eps option <https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#limits>`_ using the following parameters:
```

- timeframe = 1 seconds
- maximum = Data ingestion rate setting / number of server nodes

> The number of server nodes is automatically determined by the cloud service based on the workload. For instance, if the Data ingestion rate setting is 900 EPS and there is a cluster of 2 nodes at the current time, each node can process up to 450 events per second.

Reviewer: ```suggestion
The cloud service automatically determines the number of server nodes  based on the workload. For instance, if the ``Data ingestion rate`` setting is ``900`` EPS and the current cluster consists of 2 nodes, each node can process up to 450 events per second.
```

.. _cloud_settings_adjust:

Adjusting environment settings
------------------------------

> Managing your environment settings is crucial to meeting your evolving needs and optimizing the performance of your cloud environment. While some settings can be determined upfront, such as the number of active agents, indexed data time, archived data, and support type, it's important to note that these requirements may change over time.

Reviewer: ```suggestion
Managing your environment settings is crucial to meet your evolving needs and optimize the performance of your cloud environment. While some settings can be determined upfront, such as the number of active agents, indexed data time, archived data, and support type, note that these requirements might change over time.
```

> Additionally, advanced settings may be more challenging to determine in advance. While the interface provides recommendations based on our experience, your specific workload may differ. That's why we highly recommend deploying, monitoring, and adjusting the settings as needed to align with your evolving requirements.

Reviewer: ```suggestion
Advanced settings might be more challenging to determine in advance. While the interface provides recommendations based on our experience, your specific workload might differ. Hence, we recommend deploying, monitoring, and adjusting the settings as needed to align with your evolving requirements.
```

To effectively monitor and adapt your environment, you have the option to modify your settings by opening a support ticket. Here's how the process works:

> - Upgrading a setting: If you need to raise a setting, you will be charged a prorated amount based on the remaining time in your billing cycle. The change will be implemented immediately after the payment is made. Please note that your next billing cycle will reflect the increased cost of the enhancement.

Reviewer: ```suggestion
-  **Upgrading a setting**: If you need to raise a setting, you will be charged a prorated amount based on the remaining time in your billing cycle. The change will be implemented immediately after the payment is made. Please note that your next billing cycle will reflect the increased cost of the enhancement.
```
